From 5e3b6807b59ad9dbae0727b75a0e603f474d6451 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Thu, 31 Jan 2019 16:46:30 +0100
Subject: [PATCH] distutils/sysconfig: append
 STAGING_LIBDIR/python-sysconfigdata to sys.path

So that target configuration can be used when running native python

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>

---
 Lib/distutils/sysconfig.py | 2 ++
 Lib/sysconfig.py           | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/Lib/distutils/sysconfig.py b/Lib/distutils/sysconfig.py
index 37feae5..4774e12 100644
--- a/Lib/distutils/sysconfig.py
+++ b/Lib/distutils/sysconfig.py
@@ -444,6 +444,8 @@ def _init_posix():
         platform=sys.platform,
         multiarch=getattr(sys.implementation, '_multiarch', ''),
     ))
+    if 'STAGING_LIBDIR' in os.environ:
+        sys.path.append(os.environ['STAGING_LIBDIR']+'/python-sysconfigdata')
     _temp = __import__(name, globals(), locals(), ['build_time_vars'], 0)
     build_time_vars = _temp.build_time_vars
     global _config_vars
diff --git a/Lib/sysconfig.py b/Lib/sysconfig.py
index bf04ac5..010f9ef 100644
--- a/Lib/sysconfig.py
+++ b/Lib/sysconfig.py
@@ -417,6 +417,8 @@ def _init_posix(vars):
     """Initialize the module as appropriate for POSIX systems."""
     # _sysconfigdata is generated at build time, see _generate_posix_vars()
     name = _get_sysconfigdata_name()
+    if 'STAGING_LIBDIR' in os.environ:
+        sys.path.append(os.environ['STAGING_LIBDIR']+'/python-sysconfigdata')
     _temp = __import__(name, globals(), locals(), ['build_time_vars'], 0)
     build_time_vars = _temp.build_time_vars
     vars.update(build_time_vars)
